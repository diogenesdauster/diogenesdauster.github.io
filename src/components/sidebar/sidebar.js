import React from 'react'
import { StaticQuery, graphql } from 'gatsby'
import { Link } from 'gatsby'

export const PureSidebar = ({ data }) => {
  const menus = data.allMarkdownRemark.edges
  return (
    <div className="sidebar">
      <div className="container sidebar-sticky">
        <div className="sidebar-about">
          <h1>
            <a href="/">
              <span role="img" aria-label="emoji">
                👨🏻‍💻{' '}
              </span>
              <span className="tags">{`</`}</span>
              {`D.Dauster`}
              <span className="tags">{`>`}</span>
            </a>
          </h1>
          <p className="tagline">Code Has No Accent</p>
        </div>
        <input type="checkbox" id="menu-icon" />
        <label className="menu-label" htmlFor="menu-icon" />
        <div className="trigger">
          <ul className="sidebar-nav">
            <li>
              <Link to="/" className="sidebar-nav-item">
                Home
              </Link>
            </li>
            {menus
              ? menus.map(menu => {
                  const { name } = menu.node.frontmatter
                  const url = `${name.toLowerCase()}`
                  return (
                    <li key={name}>
                      <Link to={url} className="sidebar-nav-item">
                        {name}
                      </Link>
                    </li>
                  )
                })
              : null}
          </ul>
        </div>
        <p className="copyright">
          {' '}
          &copy; {new Date().getFullYear()}. All rights reserved.
        </p>
      </div>
    </div>
  )
}

export const Sidebar = props => (
  <StaticQuery
    query={graphql`
      query {
        allMarkdownRemark(
          filter: { frontmatter: { layout: { eq: "sidebar" } } }
          sort: { fields: frontmatter___name, order: ASC }
        ) {
          edges {
            node {
              frontmatter {
                name
              }
            }
          }
        }
      }
    `}
    render={data => <PureSidebar {...props} data={data} />}
  />
)

export default Sidebar
