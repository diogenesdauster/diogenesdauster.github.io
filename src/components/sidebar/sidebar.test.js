import React from 'react'
import renderer from 'react-test-renderer'
import { PureSidebar as Sidebar } from './sidebar'

describe('Sidebar', () =>
  it('renders correctly', () => {
    const data = {
      allMarkdownRemark: {
        edges: [
          {
            node: {
              frontmatter: {
                name: 'About',
              },
            },
          },
          {
            node: {
              frontmatter: {
                name: 'Archive',
              },
            },
          },
          {
            node: {
              frontmatter: {
                name: 'Projects',
              },
            },
          },
        ],
      },
    }
    const tree = renderer.create(<Sidebar data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
