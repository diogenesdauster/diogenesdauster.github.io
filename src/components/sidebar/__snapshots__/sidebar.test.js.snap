// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div
  className="sidebar"
>
  <div
    className="container sidebar-sticky"
  >
    <div
      className="sidebar-about"
    >
      <h1>
        <a
          href="/"
        >
          <span
            aria-label="emoji"
            role="img"
          >
            👨🏻‍💻
             
          </span>
          <span
            className="tags"
          >
            &lt;/
          </span>
          D.Dauster
          <span
            className="tags"
          >
            &gt;
          </span>
        </a>
      </h1>
      <p
        className="tagline"
      >
        Code Has No Accent
      </p>
    </div>
    <input
      id="menu-icon"
      type="checkbox"
    />
    <label
      className="menu-label"
      htmlFor="menu-icon"
    />
    <div
      className="trigger"
    >
      <ul
        className="sidebar-nav"
      >
        <li>
          <Link
            className="sidebar-nav-item"
            to="/"
          >
            Home
          </Link>
        </li>
        <li>
          <Link
            className="sidebar-nav-item"
            to="about"
          >
            About
          </Link>
        </li>
        <li>
          <Link
            className="sidebar-nav-item"
            to="archive"
          >
            Archive
          </Link>
        </li>
        <li>
          <Link
            className="sidebar-nav-item"
            to="projects"
          >
            Projects
          </Link>
        </li>
      </ul>
    </div>
    <p
      className="copyright"
    >
       
      © 
      2018
      . All rights reserved.
    </p>
  </div>
</div>
`;
